### ISSUE TITLE
"Oops; flutter has exited unexpectedly: "Bad state: No element"

### ISSUE THREAD

**Issue by MuthamilSelvam** [flutter/flutter#59050]
Hi All,

I was working with flutter on past 1 year, but it suddenly throws **"Oops; flutter has exited unexpectedly: "Bad state: No element". "**  issue for any flutter commands. i dont know why..

i removed all and reinstalled everything., 

1. GIT
2. Flutter
  Tried below 3 versions
    a. flutter_windows_1.17.1-stable
    b. flutter_windows_1.17.2-stable
    c. flutter_windows_v1.12.13+hotfix.9-stable

3. Environment variable setup (System32 path, git path, flutter sdk path), 
4. Android Studio and its Flutter and Dart plugins. 

But nothing was helped me..

## Logs

```
C:\Users\muthamil.selvam>flutter doctor

Oops; flutter has exited unexpectedly: "Bad state: No element".
Unable to generate crash report due to secondary error: Bad state: No element
please let us know at https://github.com/flutter/flutter/issues.

Oops; flutter has exited unexpectedly: "Bad state: No element".
Unable to generate crash report due to secondary error: Bad state: No element
please let us know at https://github.com/flutter/flutter/issues.
Unhandled exception:
ProcessExit: 1
#0      _handleToolError (package:flutter_tools/runner.dart:149:7)
#1      _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dar
t:86:61)
#2      _rootRunBinary (dart:async/zone.dart:1204:38)
#3      _CustomZone.runBinary (dart:async/zone.dart:1093:19)
#4      _FutureListener.handleError (dart:async/future_impl.dart:155:20)
#5      Future._propagateToListeners.handleError (dart:async/future_impl.dart:69
4:47)
#6      Future._propagateToListeners (dart:async/future_impl.dart:715:24)
#7      Future._completeError (dart:async/future_impl.dart:534:5)
#8      _AsyncAwaitCompleter.completeError (dart:async-patch/async_patch.dart:43
:15)
#9      _informUserOfCrash (package:flutter_tools/runner.dart)
#10     _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dar
t:86:61)
#11     _rootRunBinary (dart:async/zone.dart:1204:38)
#12     _CustomZone.runBinary (dart:async/zone.dart:1093:19)
#13     _FutureListener.handleError (dart:async/future_impl.dart:155:20)
#14     Future._propagateToListeners.handleError (dart:async/future_impl.dart:69
4:47)
#15     Future._propagateToListeners (dart:async/future_impl.dart:715:24)
#16     Future._completeError (dart:async/future_impl.dart:534:5)
#17     _AsyncAwaitCompleter.completeError (dart:async-patch/async_patch.dart:43
:15)
#18     _doctorText (package:flutter_tools/runner.dart)
#19     _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dar
t:86:61)
#20     _rootRunBinary (dart:async/zone.dart:1204:38)
#21     _CustomZone.runBinary (dart:async/zone.dart:1093:19)
#22     _FutureListener.handleError (dart:async/future_impl.dart:155:20)
#23     Future._propagateToListeners.handleError (dart:async/future_impl.dart:69
4:47)
#24     Future._propagateToListeners (dart:async/future_impl.dart:715:24)
#25     Future._completeError (dart:async/future_impl.dart:534:5)
#26     _AsyncAwaitCompleter.completeError (dart:async-patch/async_patch.dart:43
:15)
#27     AppContext.run (package:flutter_tools/src/base/context.dart)
#28     _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dar
t:86:61)
#29     _rootRunBinary (dart:async/zone.dart:1204:38)
#30     _CustomZone.runBinary (dart:async/zone.dart:1093:19)
#31     _FutureListener.handleError (dart:async/future_impl.dart:155:20)
#32     Future._propagateToListeners.handleError (dart:async/future_impl.dart:69
4:47)
#33     Future._propagateToListeners (dart:async/future_impl.dart:715:24)
#34     Future._completeError (dart:async/future_impl.dart:534:5)
#35     _AsyncAwaitCompleter.completeError (dart:async-patch/async_patch.dart:43
:15)
#36     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/conte
xt.dart)
#37     _asyncErrorWrapperHelper.errorCallback (dart:async-patch/async_patch.dar
t:86:61)
#38     _rootRunBinary (dart:async/zone.dart:1204:38)
#39     _CustomZone.runBinary (dart:async/zone.dart:1093:19)
#40     _FutureListener.handleError (dart:async/future_impl.dart:155:20)
#41     Future._propagateToListeners.handleError (dart:async/future_impl.dart:69
4:47)
#42     Future._propagateToListeners (dart:async/future_impl.dart:715:24)
#43     Future._completeError (dart:async/future_impl.dart:534:5)
#44     Future._asyncCompleteError.<anonymous closure> (dart:async/future_impl.d
art:582:7)
#45     _rootRun (dart:async/zone.dart:1184:13)
#46     _CustomZone.run (dart:async/zone.dart:1077:19)
#47     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:1003:
23)
#48     _microtaskLoop (dart:async/schedule_microtask.dart:43:21)
#49     _startMicrotaskLoop (dart:async/schedule_microtask.dart:52:5)
#50     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:
13)
#51     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:405:11)
#52     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:429:5)
#53     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dar
t:168:12)

```

Please help me to come out from this issue. Thanks in advance..

**Comment by TahaTesser** [comment]
Hi @MuthamilSelvam 
Can you please provide your `flutter doctor -v`?
Thank you

**Comment by MuthamilSelvam** [comment]
> Thank you

flutter doctor -v also throwing the same error which i mentioned above.

Thank you..

**Comment by TahaTesser** [comment]
Hi @MuthamilSelvam 
Can you please provide your `git --version`?

**Comment by MuthamilSelvam** [comment]
> Hi @MuthamilSelvam
> Can you please provide your `git --version`?

![image](https://user-images.githubusercontent.com/66661260/84250945-49999700-ab2a-11ea-8910-71674186cdfd.png)

**Comment by jmagman** [comment]
Would you mind trying this on the [beta channel](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels#beta), and paste the stack again?  This code recently moved around with https://github.com/flutter/flutter/pull/55614.

```
$ flutter channel beta
$ flutter upgrade
$ flutter doctor -v
```

**Comment by MuthamilSelvam** [comment]
Thank you all, Finally I resolved my issue...🤩🤩

The problem was _"ANDROID_HOME"_ environment variable has been deprecated, and I changed it as _"ANDROID_SDK_ROOT"_. And also I added the sdk path in "PATH" system environment variable.

But still issue was there, then I come to know that something happened in Android SDK. Some files may corrupted. But I am not sure.

Based on environment variable issue, i realized and just tried with new android sdk by downloading in new path and set it in system environment variable. 

On thing, don't try "_flutter config --android-sdk <sdk path>_" command. it is also caused same kind of issue for me. If you did at any cost and facing problem, then just delete _.flutter_settings_ file in the path _"C:\Users\<user-name>\AppData\Roaming\"_

And finally my issue got resolved by doing all those things.....


One funny thing happened in this R&D, was I wasted my time by reinstalled Windows OS ... 
LOL right...😂😂😂😂

Thank you all, Make all fun..! And **Be happy always**..!😃
